src/FormInput/FormInput.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+
+require("./FormInput.less");
+
+const DEFAULT_WIDTH = 400;
+
+export class FormInput extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    var wrapperClass = "FormInput";
+
+    // add additional wrapper classes
+    if (this.props.error) wrapperClass += " hasError";
+    else if (this.props.isValid) wrapperClass += " isValid";
+
+    if (this.props.readonly) {
+      wrapperClass += " readonly";
+    } else if (this.props.disabled) {
+      wrapperClass += " disabled";
+    }
+
+    // note on the upper right corner
+    var inputNote;
+    if (this.props.required) {
+      inputNote = <span className="FormInput--required">required</span>;
+    }
+    if (this.props.error) {
+      inputNote = <span className="FormInput--error">{this.props.error}</span>;
+    }
+
+    return (
+      <div className={wrapperClass}>
+        <div className="FormInput--infoRow">
+          <label className="FormInput--label" htmlFor={this.props.name}>{this.props.label}</label>
+          {inputNote}
+        </div>
+        <input
+          type="text"
+          name={this.props.name}
+          className="FormInput--input"
+          placeholder={this.props.placeholder}
+          value={this.props.value}
+          onChange={this.props.onChange}
+          readOnly={this.props.readOnly ? true : false}
+          disabled={this.props.disabled ? true : false}
```

Comment:
since readOnly and disabled are bool types, you should be able to just pass them directly